### PR TITLE
Filtered Android tasks should include mokey and pixel 7 pro in dashboard

### PR DESCRIPTION
**Pull request by jmagman**
Update the dashboard so "mokey" and "pixel_7pro"  tasks are filtered/iconed as Android.  https://github.com/flutter/flutter/issues/148085

For example, `Linux_mokey old_gallery__transition_perf` should show up in the dashboard as an Android test, but its icon and filtering counts it as a Linux test:

Screenshot 2024-09-12 at 9 57 52 AM
Screenshot 2024-09-12 at 9 57 48 AM

Also fix two unrelated warnings.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md

### CODE REVIEW

File: dashboard/lib/widgets/task_icon.dart [hunk 1]
```diff
@@ -51,6 +51,7 @@ class TaskIcon extends StatelessWidget {
     final String matchedName = qualifiedTask.task!.toLowerCase();
     final bool isWebTest = matchedName.contains('_web') || matchedName.contains('web_');
     final bool isToolTest = matchedName.contains('_tool') || matchedName.contains('tool_');
+    final bool isAndroidTest = matchedName.contains('_android') || matchedName.contains('_mokey');
```
Comment (zanderso) [inline dashboard/lib/widgets/task_icon.dart:54]: All the instances of `_android` should be gone, and I think all we have now are `_mokey` and `_pixel_7pro`.

Comment (jmagman) [reply dashboard/lib/widgets/task_icon.dart:54]: Maybe next time we do this we use "_highend" and "_lowend" or something similar so the builder names don't change.

File: dashboard/lib/main.dart [hunk 2]
```diff
@@ -8,7 +8,6 @@ import 'package:firebase_core/firebase_core.dart';
 import 'package:firebase_crashlytics/firebase_crashlytics.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter/rendering.dart';
```
Comment (jmagman) [inline dashboard/lib/main.dart:11]: Unrelated to this change but fix the warning:
```
info: The import of 'package:flutter/rendering.dart' is unnecessary because all of the used elements are also provided by the import of 'package:flutter/material.dart'. (unnecessary_import at [flutter_dashboard] lib/main.dart:11)
```

File: dashboard/test/widgets/task_grid_test.dart [hunk 3]
```diff
@@ -878,5 +878,10 @@ Future<void> expectTaskBoxColorWithMessage(WidgetTester tester, String message,
   expect(pixels!.lengthInBytes, (cellPixelArea * 4).round());
   const double padding = 4.0;
   final int rgba = pixels.getUint32((((cellPixelSize * (cellSize + padding)) + cellSize + padding).ceil()) * 4);
-  expect((rgba >> 8) | (rgba << 24) & 0xFFFFFFFF, expectedColor.value);
+  final Color actualColor = Color((rgba >> 8) | (rgba << 24) & 0xFFFFFFFF);
```
Comment (jmagman) [inline dashboard/test/widgets/task_grid_test.dart:881]: Unrelated to this change but fix the warning:
```
info: 'value' is deprecated and shouldn't be used. Use component accessors like .r or .g. (deprecated_member_use at [flutter_dashboard] test/widgets/task_grid_test.dart:881)
```

Comment (TahaTesser) [reply dashboard/test/widgets/task_grid_test.dart:881]: I've same issue in https://github.com/flutter/cocoon/pull/3901#issuecomment-2348290865. I provided some details on how it seems to be handled in the framework and tagged @gaaclarke.

**Review by zanderso — APPROVED** [review]

File: dashboard/lib/widgets/task_icon.dart (hunk 1 above)
Comment (zanderso) [reply dashboard/lib/widgets/task_icon.dart]: Changing the builder name is our current mechanism for starting a new series in SkiaPerf, which we want when we change the kind of device a benchmark runs on. We could investigate a new mechanism for that, and then we could leave the builder names the same.

File: dashboard/test/widgets/task_grid_test.dart (hunk 3 above)
Comment (jmagman) [reply dashboard/test/widgets/task_grid_test.dart:881]: @TahaTesser do you have any concerns about how I handled it here, checking `shade500`?

Comment (jmagman) [reply dashboard/test/widgets/task_grid_test.dart:881]: You make a good point though, it would be good to have `isEquivalentTo` available outside the framework for anyone else hitting this deprecation.

File: dashboard/lib/widgets/task_icon.dart (hunk 1 above)
Comment (jmagman) [reply dashboard/lib/widgets/task_icon.dart]: Or they could be different, but have some kind of stable way to identify them as Android from the name agnostic of the device hardware, so this doesn't need to be updated every time.

File: dashboard/test/widgets/task_grid_test.dart (hunk 3 above)
Comment (TahaTesser) [reply dashboard/test/widgets/task_grid_test.dart:881]: Your fix is great! This is an ongoing change so I just wanted to loop in for this edge case :)

Comment (jmagman) [reply dashboard/test/widgets/task_grid_test.dart:881]: Ah ha, https://github.com/flutter/website/pull/11132 says use `isSameColorAs` matcher, which works on this PR.

Comment (TahaTesser) [reply dashboard/test/widgets/task_grid_test.dart:881]: Thanks!
